src/abstractWrappers/Disputes.js

```diff
@@ -648,100 +650,124 @@ class Disputes extends AbstractWrapper {
-        // eslint-disable-next-line no-unused-vars
-      } catch (err) {
-        // fetching dispute will fail if it hasn't been added to the store yet. this is ok we can just not return store data
-      }
+    let appealRuledAt = []
+    let netPNK = 0
+    try {
+      const userData = await this._StoreProvider.getDisputeData(
+        arbitratorAddress,
+        disputeID,
+        account
+      )
+      if (userData.appealDraws) appealDraws = userData.appealDraws
+      if (userData.appealCreatedAt) appealCreatedAt = userData.appealCreatedAt
+      if (userData.appealDeadlines) appealDeadlines = userData.appealDeadlines
+      if (userData.appealRuledAt) appealRuledAt = userData.appealRuledAt
+      if (userData.netPNK) netPNK = userData.netPNK
+      // eslint-disable-next-line no-unused-vars
+    } catch (err) {
+      // Fetching a dispute will fail if it hasn't been added to the store yet. This is ok, we can just not return store data
     }
 
-    // get evidence
-    const evidence = await this.getEvidenceForArbitrableContract(
-      arbitrableContractAddress
-    )
-
-    const firstSession = dispute.firstSession
+    // Build juror info and ruling arrays, indexed by appeal number
     const lastSession = dispute.firstSession + dispute.numberOfAppeals
-    // NOTE arrays indexed by appeal number
-    const appealRulings = []
     const appealJuror = []
-    for (let appeal = 0; appeal <= lastSession - firstSession; appeal++) {
-      // get ruling for appeal. Note appeal 0 is first session
-      const ruling = await this._Arbitrator.currentRulingForDispute(
-        arbitratorAddress,
-        disputeId,
-        appeal
-      )
-
-      appealRulings[appeal] = {
-        ruling,
-        voteCounter: dispute.voteCounters[appeal],
-        ruledAt: appealRuledAt[appeal],
-        deadline: appealDeadlines[appeal]
-      }
+    const appealRulings = []
+    for (let appeal = 0; appeal <= dispute.numberOfAppeals; appeal++) {
+      const isLastAppeal = dispute.firstSession + appeal === lastSession
 
-      const baseFee = dispute.arbitrationFeePerJuror
+      // Get appeal data
       const draws = appealDraws[appeal] || []
       let canRule = false
-      if (firstSession + appeal === lastSession && draws.length > 0) {
-        canRule = await this._Arbitrator.canRuleDispute(
+      let canRepartition = false
+      let canExecute = false
+      let ruling
+      const promises = [
+        this._Arbitrator.currentRulingForDispute(
           arbitratorAddress,
-          disputeId,
-          draws,
-          account
+          disputeID,
+          appeal
         )
+      ]
+
+      // Extra info for the last appeal
+      if (isLastAppeal) {
+        if (draws.length > 0)
+          promises.push(
+            this._Arbitrator.canRuleDispute(
+              arbitratorAddress,
+              disputeID,
+              draws,
+              account
+            )
+          )
+
+        if (session && period)
```

Comment:
I don't understand why we need these passed in. Why not just fetch from contract vs relying on user to determine period and session beforehand?

Reply:
Because most of the time you are fetching a list of disputes, it would make this even slower to refetch the same data for every dispute.